compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -161,32 +191,36 @@ object Objects:
    *
    * @param owner The static object whose initialization creates the array.
    */
-  case class OfArray(owner: ClassSymbol, regions: Regions.Data)(using @constructorOnly ctx: Context)
-  extends Ref(valsMap = mutable.Map.empty, varsMap = mutable.Map.empty, outersMap = mutable.Map.empty):
+  case class OfArray(owner: ClassSymbol, regions: Regions.Data)(using @constructorOnly ctx: Context) extends ValueElement:
```

Comment:
@liufengyun Now that `OfArray` is not a subtype of `Ref`, do we need to explicitly handle `OfArray` in `select` and `assign`?

Reply:
As discussed in the meeting, for `select` and `assign`, we can issue an internal error (it should not be reachable). For `call`, we can move the code currently in `ref: Ref` to `array: OfArray`.